src/components/panel/panel.js

```diff
@@ -1101,69 +1129,263 @@ MdPanelPosition.prototype.top = function(opt_top) {
 
 
 /**
- * Gets the value of `top` for the panel.
- * @returns {string}
+ * Sets the value of `bottom` for the panel.
+ * @param {string=} opt_bottom Value of `bottom`. Defaults to '0'.
+ * @returns {MdPanelPosition}
  */
-MdPanelPosition.prototype.getTop = function() {
-  return this._top;
+MdPanelPosition.prototype.bottom = function(opt_bottom) {
+  this._bottom = opt_bottom || '0';
+  return this;
 };
 
 
 /**
- * Sets the value of `bottom` for the panel.
- * @param {string=} opt_bottom Value of `bottom`. Defaults to '0'.
+ * Sets the value of `left` for the panel.
+ * @param {string=} opt_left Value of `left`. Defaults to '0'.
  * @returns {MdPanelPosition}
  */
-MdPanelPosition.prototype.bottom = function(opt_bottom) {
-  this._bottom = opt_bottom || '0';
+MdPanelPosition.prototype.left = function(opt_left) {
+  this._left = opt_left || '0';
   return this;
 };
 
 
 /**
- * Gets the value of `bottom` for the panel.
- * @returns {string}
+ * Sets the value of `right` for the panel.
+ * @param {string=} opt_right Value of `right`. Defaults to '0'.
+ * @returns {MdPanelPosition}
  */
-MdPanelPosition.prototype.getBottom = function() {
-  return this._bottom;
+MdPanelPosition.prototype.right = function(opt_right) {
+  this._right = opt_right || '0';
+  return this;
 };
 
 
 /**
- * Sets the value of `left` for the panel.
- * @param {string=} opt_left Value of `left`. Defaults to '0'.
+ * Sets element for relative positioning.
+ * @param {string|!Element|!angular.JQLite} element Query selector,
+ *     DOM element, or angular element to set the panel relative to.
  * @returns {MdPanelPosition}
  */
-MdPanelPosition.prototype.left = function(opt_left) {
-  this._left = opt_left || '0';
+MdPanelPosition.prototype.relativeTo = function(element) {
+  this._absolute = false;
+  this._relativeToRect = getBoundingClientRect(getElement(element));
   return this;
 };
 
 
 /**
- * Gets the value of `left` for the panel.
- * @returns {string}
+ * Sets the x position for the panel relative to another element.
+ * xPosition must be one of the MdPanelPosition.xPosition values.
+ *
+ * @param {string} xPosition
+ * @returns {MdPanelPosition}
  */
-MdPanelPosition.prototype.getLeft = function() {
-  return this._left;
+MdPanelPosition.prototype.withPanelXPosition = function(xPosition) {
+  if (!this._relativeToRect) {
+    throw new Error('withPanelXPosition can only be used with relative' +
+        'positioning. Set relativeTo first.');
+  }
+
+  var validPosition = false;
+  for (var key in MdPanelPosition.xPosition) {
+    if (MdPanelPosition.xPosition[key] === xPosition) {
+      this._xPosition = xPosition;
+      validPosition = true;
+      break;
+    }
+  }
+
+  if (!validPosition) {
+    throw new Error('withPanelXPosition only accepts the following values:\n' +
+        'center | align-left | align-right | align-start | align-end | ' +
+        'offset-left | offset-right | offset-start | offset-end');
```

Comment:
Isn't that a bit hard coded? What do you think about generating those possible values from the array?